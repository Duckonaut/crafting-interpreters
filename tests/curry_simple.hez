fn add_three(a, b, c) {
    return a + b + c;
}

fn add_three(a, b, c) {
    return a + b + c;
}

fn curried_add_three(a) {
    fn ca(b) {
        fn cb(c) {
            return a + b + c;
        }
        return cb;
    }
    return ca;
}

println("curried");
println(add_three(1, 2, 3));
println(curried_add_three(1)(2)(3));
var partial = curried_add_three(1);
println(partial(2)(3));
