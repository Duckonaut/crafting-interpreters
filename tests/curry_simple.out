curried
6
6
6
